### PR TITLE
Update templates to use new ResourceOptions api.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/pulumi/pulumi-kubernetes/issues/595

### CODE REVIEW

File: pkg/gen/python-templates/yaml.py.mustache [hunk 1]
```diff
@@ -64,11 +64,7 @@ class ConfigFile(pulumi.ComponentResource):
         else:
             text = _read_file(file_id)
 
-        if opts is not None:
-            _opts = deepcopy(opts)
-            _opts.parent = self
-        else:
-            _opts = pulumi.ResourceOptions(parent=self)
+        _opts = pulumi.ResourceOptions.merge(opts, pulumi.ResourceOptions(parent=self))
```
Comment (lblackstone) [inline pkg/gen/python-templates/yaml.py.mustache]: Might want to rename this to `opts` for consistency (or vice versa).

Comment (CyrusNajmabadi) [reply pkg/gen/python-templates/yaml.py.mustache]: yup yup

File: tests/examples/python/helm/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
-pulumi>=0.17.1,<2.0.0
-pulumi_random>=0.5.6,<0.6.0
+pulumi>=1.0.0b3,<2.0.0
+pulumi_random>=1.0.0b1,<2.0.0
```
Comment (lblackstone) [inline tests/examples/python/helm/requirements.txt]: I think this should actually be `pulumi-random` https://pypi.org/project/pulumi-random/1.0.0b1/

Comment (CyrusNajmabadi) [reply tests/examples/python/helm/requirements.txt]: Doesn't seem like it.  that causes:

```
pulumi:providers:random default_1_0_1  error: no resource plugin 'random-v1.0.1' found in the workspace or on your $PATH, install the plugin using `pulumi plugin install resource random v1.0.1`

[ tes/tests/examples/python/helm ]  +  pulumi:pulumi:Stack helm-p-it-travis-job-helm-1fae8433 create 1 warning

[ tes/tests/examples/python/helm ]     pulumi:providers:random default_1_0_1  1 error
```

**Pull request merged**